gateway-service-knoxtoken/src/main/java/org/apache/knox/gateway/service/knoxtoken/TokenResource.java

```diff
@@ -729,6 +743,19 @@ private Response getAuthenticationToken() {
     return Response.ok().entity("{ \"Unable to acquire token.\" }").build();
   }
 
+  private void addArbitraryTokenMetadata(TokenMetadata tokenMetadata) {
+    final String metadataPrefix = "md_";
+    final Enumeration<String> paramNames = request.getParameterNames();
+    while (paramNames.hasMoreElements()) {
+      final String paramName = paramNames.nextElement();
+      if (paramName.startsWith(metadataPrefix)) {
+        final String metadataName = paramName.replace(metadataPrefix, "");
```

Comment:
Is this less expensive than `paramName.substring(metadataPrefix.length() + 1)` ? What if the paramName value itself includes the metadataPrefix String (e.g., md_MyMetadata_md_xyz)

Reply:
Nice catch (both the performance observation and the MD naming). I'll fix it soon.

Reply:
Fixed.